cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -553,9 +541,6 @@ public void getStatuses(HttpRequest request, HttpResponder responder,
         ProgramStatus programStatus = lifecycleService.getProgramStatus(programId);
         statuses.add(new BatchProgramStatus(
           program, HttpResponseStatus.OK.getCode(), null, programStatus.name()));
-      } catch (BadRequestException e) {
-        statuses.add(new BatchProgramStatus(
```

Comment:
so I split ``findRuntimeInfo(ProgramId, runId)`` into two methods: ``findRuntimeInfo(ProgramId)`` and ``findRuntimeInfo(ProgramId, runId)``. Only ``findRuntimeInfo(ProgramId, runId)`` throws ``BadRequestException``. The method only throws BadRequestException when a runId is provided, but is invalid. Didn't make sense to have it in the throws clause when we request runtime info without a runId